### ISSUE TITLE
IMAP mail not getting deleted on gmail.

### ISSUE THREAD

**Issue by Nodens-** [getmail6/getmail6#107]
```
getmail version 6.18.5
Copyright (C) 1998-2021 Charles Cazabon and others. Licensed under GNU GPL version 2.
getmail configuration:
  getmail version 6.18.5
  Python version 3.10.1 (main, Dec  9 2021, 00:00:00) [GCC 11.2.1 20211203 (Red Hat 11.2.1-7)]
  retriever:  SimpleIMAPSSLRetriever(ca_certs="None", certfile="None", getmaildir="/var/mail/vmail/.getmail", imap_on_delete="None", imap_search="None", keyfile="None", mailboxes="('[Gmail]/All Mail',)", password="*", password_command="()", port="993", record_mailbox="True", server="imap.gmail.com", ssl_cert_hostname="None", ssl_ciphers="None", ssl_fingerprints="()", ssl_version="None", timeout="180", use_cram_md5="False", use_kerberos="False", use_peek="True", use_xoauth2="False", username="email@gmail.com")
  destination:  MDA_external(allow_root_commands="False", arguments="('-e', '-d', 'email@gmail.com')", command="deliver", group="None", ignore_stderr="False", path="/usr/libexec/dovecot/deliver", unixfrom="False", user="None")
  options:
    delete : True
    delete_after : 0
    delete_bigger_than : 0
    delivered_to : False
    fingerprint : False
    logfile : logfile(filename="/var/log/getmail.log")
    max_bytes_per_session : 0
    max_message_size : 0
    max_messages_per_session : 0
    message_log : /var/log/getmail.log
    message_log_syslog : False
    message_log_verbose : False
    netrc_file : None
    read_all : False
    received : False
    skip_imap_fetch_size : False
    use_netrc : False
    verbose : 1
```

```
[retriever]
type = SimpleIMAPSSLRetriever
server = imap.gmail.com
port = 993
username = email@gmail.com
password = apassword
mailboxes = ("[Gmail]/All Mail",)

[destination]
type = MDA_external
path = /usr/libexec/dovecot/deliver
arguments = ("-e", "-d", "email@gmail.com")

[options]
read_all = false
delete = true
delivered_to = false
received = false
message_log = /var/log/getmail.log
message_log_syslog = false
```

If I add `move_on_delete = "[Gmail]/Trash"` option in `[retriever]` it gets moved to the trash directory but I actually want mail deleted on the spot for security reasons. I did try disabling autoexpunge on gmail settings and enabling the option to delete immediately when client sends `EXPUNGE` but that had no effect either which makes me think getmail6 does not send `EXPUNGE`?

**Comment by rpuntaie** [comment]
getmail: If `delete = true`, then `IMAP4.expunge()`, i.e. (`EXPUNGE`).

I think it is a google setting,
maybe
https://support.google.com/mail/thread/5732292/auto-expunge-on-and-move-message-to-trash-rather-than-archive?hl=en

Please report back.

**Comment by Nodens-** [comment]
Like I said in the original comment I have already tried this setting. `delete=true` just doesn't work at all. Not with autoexpunge on, not with autoexpunge off and setting to delete on client sent expunge. The only option that works is `move_on_delete = "[Gmail]/Trash"` coupled with `delete = true` but that's obviously not deleting...

**Comment by rpuntaie** [comment]
What if you add `[Gmail]/Trash` to `mailboxes`?

**Comment by Nodens-** [comment]
I just tried it. That caused mail to be downloaded twice with autoexpunge on, on gmail settings, and no deletion. With autoexpunge off and delete immediately setting, it downloaded the mail twice and then deleted it. So that's not optimal either.

I did manage to find something that works though:
`mailboxes = ("[Gmail]/Inbox", "[Gmail]/Spam")`

Using this combination of mailboxes instead of `"[Gmail]/All Mail"`, along with autoexpunge off and delete immediately, on gmail side of things, mail gets deleted properly and spam mail gets downloaded as well. Making an educated guess it seems that it's a gmail limitation and the "All Mail" directory, which holds all mail, regardless of filtering on their side, can not be deleted from (but apparently can be moved to Trash). I do not know if anything can be done about that from getmail6 side of things, but since gmail now allows downloading of spam directory without the use of `All Mail` I have a good working solution.

My use case is providing centralized IMAP access via dovecot, on my own MX servers, for external addresses too. Which obviously allows for centralized backup and better security for storage at rest (if an external email account gets compromised, they don't get 2000 emails at rest).
